judgels-backends/judgels-commons/judgels-fs/src/main/java/judgels/fs/local/LocalFileSystem.java

```diff
@@ -106,21 +107,32 @@ public String getPrivateFileUrl(Path filePath) {
     }
 
     @Override
-    public void uploadZippedFiles(Path dirPath, InputStream content, boolean includeDirectory) {
+    public void uploadZippedFiles(Path dirPath, InputStream content) {
         try {
             File destDir = baseDir.resolve(dirPath).toFile();
             byte[] buffer = new byte[4096];
             int entries = 0;
             long total = 0;
+
+            Set<String> directories = Sets.newHashSet();
```

Comment:
I think this whole logic can be simplified by always taking `FilenameUtils.getName(ze.getName())` as the filename for all non-directory entries. Something like (pseudocode):

```
if (ze.isDirectory()) {
  continue;
}

String filename = FilenameUtils.getName(ze.getName());
File file = new File(destDir, filename);

// etc
```

What do you think?